pallets/handles/src/tests/handle_creation_tests.rs

```diff
@@ -293,3 +293,33 @@ fn claim_handle_fails_when_handle_contains_unsupported_unicode_characters() {
 		);
 	});
 }
+
+#[test]
+fn claim_handle_with_max_bytes_should_get_correct_display_handle() {
+	new_test_ext().execute_with(|| {
+		let alice = sr25519::Pair::from_seed(&[0; 32]);
+		let expiration = 100;
+		// use 4 bytes character to build a handle with 32 bytes
+		let handle = "𝔸".repeat(8);
+		let (payload, proof) =
+			get_signed_claims_payload(&alice, handle.as_bytes().to_vec(), expiration);
+		assert_ok!(Handles::claim_handle(
+			RuntimeOrigin::signed(alice.public().into()),
+			alice.public().into(),
+			proof,
+			payload.clone()
+		));
+		let msa_id = MessageSourceId::decode(&mut &alice.public().encode()[..]).unwrap();
+		let handle = Handles::get_handle_for_msa(msa_id);
+		assert!(handle.is_some());
+		let handle_result = handle.unwrap();
+		assert_eq!(handle_result.base_handle, "𝔸".repeat(8).as_bytes().to_vec());
+		assert!(handle_result.suffix > 0);
+		let display_handle = "𝔸".repeat(8) + "." + &handle_result.suffix.to_string();
+		let display_handle_vec = display_handle.as_bytes().to_vec();
+		let msa_id_from_state =
+			Handles::get_msa_id_for_handle(display_handle_vec.try_into().unwrap());
+		assert!(msa_id_from_state.is_some());
+		assert_eq!(msa_id_from_state.unwrap(), msa_id);
+	});
+}
```

Comment:
This test claims all 32b of allowed handle space and check the integrity of state especially checking if DisplayHandle is correctly formed